pydantic/types.py

```diff
@@ -858,6 +863,40 @@ def __get_pydantic_core_schema__(
         def __repr__(self) -> str:
             return 'NaiveDatetime'
 
+    class PastDatetime:
```

Comment:
It's not a big deal, but if there's no reason not to, I would suggest making this (and `FutureDatetime`) a dataclass (see the items in `pydantic/annotated_arguments.py`) as it slightly improves the repr when displayed instantiated and included in an annotation.

Reply:
I've added a commit and make this and the other date and datime related types(`PastDate`, `FutureDate`, `AwareDatetime`, ..) a dataclass

Reply:
dataclass change reverted